src/orange/BoundingBoxUtils.cc

```diff
@@ -0,0 +1,98 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/BoundingBoxUtils.cc
+//---------------------------------------------------------------------------//
+#include "BoundingBoxUtils.hh"
+
+#include "corecel/Assert.hh"
+#include "corecel/math/ArrayOperators.hh"
+#include "corecel/math/NumericLimits.hh"
+
+#include "transform/Transformation.hh"
+#include "transform/Translation.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Calculate the bounding box of a translated box.
+ */
+BBox calc_transform(Translation const& tr, BBox const& a)
+{
+    CELER_EXPECT(a);
+    return {tr.transform_up(a.lower()), tr.transform_up(a.upper())};
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculate the bounding box of a transformed box.
```

Comment:
I think you should emphasis this a bit more. This isn't transforming a bbox --- it's transforming a bbox and then finding an *axes-aligned* bbox around the translated bbox. In other words, this is doing something quite different from  other `calc_transform` functions in the code.